matchzoo/engine/base_model.py

```diff
@@ -21,6 +21,53 @@ class BaseModel(abc.ABC):
     BACKEND_WEIGHTS_FILENAME = 'backend_weights.h5'
     PARAMS_FILENAME = 'params.dill'
 
+    class EvaluateOnCall(keras.callbacks.Callback):
+        def __init__(self, x, y,
+                     metrics: typing.Union[list, str, engine.BaseMetric],
+                     valid_steps=3,
+                     batch_size: int = 32):
+            super().__init__()
+            self._x = x
+            self._y = y
+            self._metrics = metrics
+            self._valid_steps = valid_steps
+            self._batch_size = batch_size
+
+        def on_epoch_end(self, epoch, logs={}):
+            if epoch % self._valid_steps:
+                return logs
+            val_x = self._x
+            val_y = self._y
+            val_logs = self.model.evaluate(x=val_x, y=val_y,
+                                                batch_size=self._batch_size,
+                                                verbose=0)
+            if not isinstance(val_logs, list):
+                val_logs = [val_logs]
+            val_logs = {name: val for name, val in
+                    zip(self.model.metrics_names, val_logs)}
+            dataframe = None
+            for metric in self._metrics:
+                if isinstance(metric, engine.BaseMetric):
+                    if dataframe is None:
+                        y_pred = self.model.predict(val_x,
+                                                    batch_size=self._batch_size
+                                                    ).reshape((-1,))
+                        data = {
+                            'id': val_x['id_left'].tolist(),
+                            'true': val_y.tolist(),
```

Comment:
use `label` instead of `true`?

Reply:
It's consistent with my `evaluate`. I will change both of them into `y_true`, and `y_pred` later.